### PR TITLE
fix: fallback to `eth_sign`

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves https://github.com/5afe/safe-support/issues/191

## How this PR fixes it

If `eth_signTypedData` is not supported by the current wallet (e.g. Ledger), `eth_sign` is used instead to sign off-chain messages (if supported by the Safe version).

## How to test it

Try to sign an EIP-712 message off-chain with a Ledger and observe a successful message creation.

## Screenshots

off-chain-ledger

## Checklist
* [x] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Review by schmanu — APPROVED** [review]
Nice work!
Glad we figured this one out! 💪

File: src/utils/safe-messages.ts [hunk 1]
```diff
@@ -67,3 +72,38 @@ export const supportsEIP1271 = ({ version, fallbackHandler }: SafeInfo): boolean
...
+export const tryOffChainMsgSigning = async (
+  signer: providers.JsonRpcSigner,
+  safe: SafeInfo,
+  message: SafeMessage['message'],
+): Promise<string> => {
+  const signingMethods = getSupportedSigningMethods(safe.version)
+
+  for await (const [i, signingMethod] of signingMethods.entries()) {
+    try {
+      if (signingMethod === 'eth_signTypedData') {
```
Comment (katspaugh) [inline src/utils/safe-messages.ts:85]: Should we also try v3/v4? Or Ledger supports none?

Comment (schmanu) [reply src/utils/safe-messages.ts:85]: Ledger supports none and `ethers` also only knows one. 
If you check the `safe-core-sdk` all of these variants will call the same function in ethers. It only makes a difference when using the `web3` adapter.

**Comment by francovenica** [comment]
LGTM

Used the mock app for EIP-1271 test Dapp, signed a 712 with ledger off chain message. I had no issues
image

https://eip_1271_eth_sign--walletweb.review-wallet-web.5afe.dev/transactions/messages?safe=gor%3A0xEF7315E6F421ea2b583aE1625E256E559726515c

**Pull request merged**